src/commands/lint/lint.s3.paths.ts

```diff
@@ -0,0 +1,70 @@
+import { command, positional, string } from 'cmd-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { config, registerCli, verbose } from '../common.js';
+import { imageryBucketNames, imageryCrs, imageryProducts, regions } from './lint.constants.js';
+
+export const commandLintInputs = command({
+  name: 'lint-inputs',
+  description: 'Pretty print JSON files',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+    path: positional({
+      type: string,
+      displayName: 'path',
+      description: 'JSON file to load inputs from, must be a JSON Array',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    logger.info('LintInputs:Start');
+    if (!args.path) {
+      throw new Error('No path Provided');
+    }
+    logger.info({ target: args.path }, 'LintInputs:Info');
+    lintPath(args.path);
+    logger.info({ duration: performance.now() - startTime }, 'LintInputs:Done');
+  },
+});
+
+export function lintPath(path: string): void {
+  const pathArray: string[] = path.replace('s3://', '').split('/');
+  const bucket: string | undefined = pathArray[0];
+
+  if (bucket == null) {
+    throw new Error(`No Bucket Specified: ${path}`);
+  }
+
+  if (imageryBucketNames.includes(bucket)) {
+    lintImageryPath(pathArray);
+  } else {
+    throw new Error(`Bucket not bucket list: ${bucket}`);
+  }
+}
+
+export function lintImageryPath(pathArray: string[]): void {
+  const region: string | undefined = pathArray[1];
+  const dataset: string | undefined = pathArray[2];
+  const product: string | undefined = pathArray[3];
+  const crs: string | undefined = pathArray[4];
+
+  // lint target path
+  if (region == null || dataset == null || product == null || crs == null) {
+    throw new Error(`Missing key from path: s3://${pathArray.join('/')}`);
```

Comment:
It might be good to use custom error classes for all the validation errors:

- The tests could each easily check for the right error class, making them stricter.
- You could forward only the relevant inputs to the error class, and have the error class deal with formatting a message.

So in this line you might end up with `throw new MissingKeyFromPathError(pathArray);`.

Reply:
> So in this line you might end up with throw new MissingKeyFromPathError(pathArray);.

Subclassing in javascript is not super used mostly because the instance of checker is somewhat hit and miss, generally everything just throws a standard error with codes, eg a `fs.readFile('foo')` throws a `Error` with code `ENOENT`

Reply:
If we don't want to use subclassing then the tests should at least check the error message, to make sure it's catching the relevant error and not something else.

Reply:
Yes what error is thrown needs to be checked.

Simplest way is to use a RegExp in the test

```typescript
assert.throws(() => new Error('Hello'), /^Error: Hello$/)
```